### PR TITLE
Commit and push created tests

### PR DESCRIPTION
**Pull request by roxspring**
- Extracted `cover.install(..)` function to manage installation
- Extracted `cover.activate()` function to manage activation
- Added `cover.createPreflight()` and `cover.fixBuildIfNeeded(..)` functions to manage build refactorings
- Added `cover.create(..)` to manage test creation

### CODE REVIEW

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Maven Project / Test  (ubuntu-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5889157380/attempts/1)
- Commit: `f23161a92a9421a2d81f48d4bdcec3b7b64d5977`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Maven Project / Test  (macos-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5889157380/attempts/1)
- Commit: `f23161a92a9421a2d81f48d4bdcec3b7b64d5977`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Gradle Project / Test  (ubuntu-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5889157379/attempts/1)
- Commit: `f23161a92a9421a2d81f48d4bdcec3b7b64d5977`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Gradle Project / Test  (macos-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5889157379/attempts/1)
- Commit: `f23161a92a9421a2d81f48d4bdcec3b7b64d5977`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Gradle Project / Test  (windows-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5889157379/attempts/1)
- Commit: `f23161a92a9421a2d81f48d4bdcec3b7b64d5977`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Maven Project / Test  (ubuntu-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890250623/attempts/1)
- Commit: `3cee0dff66c5d76ae210f7e14f1c2ff1189f881c`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Maven Project / Test  (windows-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890250623/attempts/1)
- Commit: `3cee0dff66c5d76ae210f7e14f1c2ff1189f881c`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Gradle Project / Test  (ubuntu-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890250624/attempts/1)
- Commit: `3cee0dff66c5d76ae210f7e14f1c2ff1189f881c`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Gradle Project / Test  (windows-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890250624/attempts/1)
- Commit: `3cee0dff66c5d76ae210f7e14f1c2ff1189f881c`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Gradle Project / Test  (macos-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890250624/attempts/1)
- Commit: `3cee0dff66c5d76ae210f7e14f1c2ff1189f881c`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Maven Project / Test  (ubuntu-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890269329/attempts/1)
- Commit: `1f5b3c85ddc9e3b421914f52f043d07bd48060bc`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Maven Project / Test  (macos-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890269329/attempts/1)
- Commit: `1f5b3c85ddc9e3b421914f52f043d07bd48060bc`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Gradle Project / Test  (ubuntu-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890269335/attempts/1)
- Commit: `1f5b3c85ddc9e3b421914f52f043d07bd48060bc`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Gradle Project / Test  (macos-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890269335/attempts/1)
- Commit: `1f5b3c85ddc9e3b421914f52f043d07bd48060bc`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Gradle Project / Test  (windows-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890269335/attempts/1)
- Commit: `1f5b3c85ddc9e3b421914f52f043d07bd48060bc`

:construction: _This comment is under construction and will be updated on completion._ :construction:

**Comment by roxspring** [comment]
### Diffblue Cover

- Run: [Test: Maven Project / Test  (windows-latest)](https://github.com/diffblue/cover-github-action/actions/runs/5890269329/attempts/1)
- Commit: `1f5b3c85ddc9e3b421914f52f043d07bd48060bc`

:construction: _This comment is under construction and will be updated on completion._ :construction:

File: src/cover.ts [hunk 1]
```diff
@@ -38,6 +39,51 @@ export async function activate(): Promise<void> {
...
+    await fixBuild(status)
+  }
+}
+
+/**
+ * Runs `dcover fix-build`.
+ *
+ * @param status the status to be updated and saved.
+ */
+export async function fixBuild(status: Status): Promise<void> {
```
Comment (NlightNFotis) [inline src/cover.ts:75]: I understand that this is gearing up for a demo, so probably not worth doing right now, but potentially something for the back of our minds.

We're not doing any error detection/recovery after the `exec.exec`, nor are we returning any useful values to clients to determine whether the operation has succeeded or failed (`Promise<void>`).

Would it make sense to add a ticket or a `TODO` to revisit error detection/reporting in the future?

**Pull request merged**